middle_end/flambda2/types/meet_and_join_new.ml

```diff
@@ -565,16 +492,27 @@ and meet_expanded_head0 env (descr1 : ET.descr) (descr2 : ET.descr) :
 and meet_head_of_kind_value env (head1 : TG.head_of_kind_value)
     (head2 : TG.head_of_kind_value) : TG.head_of_kind_value meet_result =
   match head1, head2 with
-  | ( Variant { blocks = blocks1; immediates = imms1; is_unique = is_unique1 },
-      Variant { blocks = blocks2; immediates = imms2; is_unique = is_unique2 } )
-    ->
+  | ( Variant
+        { blocks = blocks1;
+          immediates = imms1;
+          extensions = extensions1;
+          is_unique = is_unique1
+        },
+      Variant
+        { blocks = blocks2;
+          immediates = imms2;
+          extensions = extensions2;
+          is_unique = is_unique2
+        } ) ->
     (* Uniqueness tracks whether duplication/lifting is allowed. It must always
        be propagated, both for meet and join. *)
     let is_unique = is_unique1 || is_unique2 in
     map_result
-      ~f:(fun (blocks, immediates) ->
-        TG.Head_of_kind_value.create_variant ~is_unique ~blocks ~immediates)
-      (meet_variant env ~blocks1 ~imms1 ~blocks2 ~imms2)
+      ~f:(fun (blocks, immediates, extensions) ->
+        TG.Head_of_kind_value.create_variant ~is_unique ~blocks ~immediates
```

Comment:
This `meet_variant` is now its own function, I think we could include this `map_result` when creating a `New_value` there.

Reply:
We could, but I think the interface is cleaner this way. Most (all ?) `meet` functions have the same types in their parameters and returns, and if we moved `create_variant` down then `meet_variant` would take individual variant fields but return a `TG.head_of_kind_value meet_result`, which feels weird.